### PR TITLE
Avoid running all checks when `aspell-dict.txt` was changed

### PR DESCRIPTION
**Pull request by AVMusorin**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Avoid running all checks when `aspell-dict.txt` was changed

For example in https://github.com/ClickHouse/ClickHouse/pull/59350 after changing `aspell-dict.txt` all tests were launched.

cc @Felixoid

### CODE REVIEW

**Review by alexey-milovidov — APPROVED** [review]

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit 3339a970144400a824262812648968847d11953a with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/59496/3339a970144400a824262812648968847d11953a/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/59496/3339a970144400a824262812648968847d11953a/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

**Comment by Felixoid** [comment]
@maxknv I am sure it should be in the `ci_config.py`, right?

PRInfo methods aren't used

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
@Felixoid 

If some methods are not used, remove them.
If you want to keep them - write comments.

**Comment by Felixoid** [comment]
All the methods `PRInfo.has_changes_*` are not used and are fine to remove.

The change should be done into https://github.com/AVMusorin/ClickHouse/blob/skip-tests-for-docs/tests/ci/ci_config.py#L809

**Comment by maxknv** [comment]
> @maxknv I am sure it should be in the `ci_config.py`, right?
> 
> PRInfo methods aren't used

- There is a method has_changes_in_documentation_only() that is used now. if it returns True we remove all checks in a CI  run except style and docs. So technically proposed change does work and with it CI just skips most of the jobs.

- There is a link to PR there this change would help to avoid unnecessary work in ci. So the idea makes sense.

- Mentioned PR's commit (where aspell-dict.txt is changed) is definitely the case when all jobs can be reused and skipped, but apparently it didn't happen there due to GH's "merge-commit approach" and because there were no job statuses to reuse at the moment of CI launch.

So, in conclusion: this is a good idea, although I would implement it a little differently because _aspell-dict.txt_ has nothing to do with documentation (and we don't even need to run docs here) and I believe there is a bunch of other files we might want to handle the same way (don't run the whole ci if changes only happen to those files). I think we need a new method that will check if there are changes only in such files, and if it is True - then apply the "don't test" behavior for ci (style check only). What do you think @Felixoid?

I'm ok to merge it as is now and implement the above later.

**Comment by AVMusorin** [comment]
Thank you for your comments. 

> I think we need a new method that will check if there are changes only in such files, and if it is True - then apply the "don't test" behavior for ci (style check only).

That sounds great to me. I can return to this PR soon and finish it with your suggestion. At least I am not satisfied with the current solution (add a new file in `has_changes_in_documentation_only`) and I am sure that it can be done better.

**Comment by maxknv** [comment]
@AVMusorin I just realised (due to another PR #60053) that this aspell-dict.txt usually changed alongside with docs update. Now I think it's absolutely correct change and we need it as is. Forget about my previous comment :)
